### ISSUE TITLE
YouTube Upload for Backup

### ISSUE THREAD

**Issue by MayhemBill** [MrBrax/LiveStreamDVR#233]
The short version is I am using this to automatically download the Streams of my communities affiliated streamers (affiliated to the community, not just twitch), and I would like to be able to automatically upload the streams to YouTube, as I am currently doing them manually. would this be possible to add in?

**Comment by MrBrax** [comment]
it was something i was going to add, but to get access to that api you need to have the channel go through a review process by youtube, which for dev purposes i'm not sure i want to do, but they might have changed that

**Comment by MrBrax** [comment]
https://developers.google.com/youtube/v3/docs/videos/insert

**Comment by MayhemBill** [comment]
So from reading that, only the channel that is being uploaded to has to go through the audit process, and it seems its more for the content of the channel itself and not the application that is being used.

**Comment by MrBrax** [comment]
yeah that's pretty much it

i gotta do some more research on how to do the login flow, i'll see what i can do

**Comment by MrBrax** [comment]
finally got the bare minimal stuff working, does need some self-configuring of apps and that in the cloud console though

**Comment by MayhemBill** [comment]
I'll swap over to the develop on compose and see if it works

**Comment by MayhemBill** [comment]
So I can see where to put in the Client ID and secret, but I can't see where to upload the videos to, or how to force older videos to get uploaded, so I'm assuming that's not possible yet.

Also I am unable to figure out where to get a client ID and secret, Google only has API keys.

**Comment by MayhemBill** [comment]
okay, got it mostly working, but when I try to upload a video it throws up a prompt and says "Insufficient Permission", but as far as I can tell I got everything setup correctly.

**Comment by MayhemBill** [comment]
small update: it seems to upload the video anyways, but makes it a private video immediately.

**Comment by MayhemBill** [comment]
`YouTube <ERROR> Could not verify video: Error: Insufficient Permission`

this is the error I get in console

**Comment by MrBrax** [comment]
yeah it's the verification check that fetches the video info to see if it got uploaded that fails

**Comment by MrBrax** [comment]
private video is hardcoded right now, i should probably make that a toggle for those who actually have an audited channel

**Comment by MrBrax** [comment]
exporting from files manager is now on develop

**Comment by CappiSteijns** [comment]
> exporting from files manager is now on develop

Is it possible to have the vods automatically uploaded to youtube after recording has finished? 

For now I can only manually upload to youtube. There is an option for auto export but that only allows for "file" and "sftp". 😃

**Comment by MrBrax** [comment]
![1662320895744](https://user-images.githubusercontent.com/1517911/188331028-bb30ca91-b57d-4971-9659-0e2607b6a77e.png)
this should be implemented...

auto upload to youtube isn't possible as far as i know, as the token you authenticate from your account expires within like 2 hours or something like that

**Comment by MrBrax** [comment]
so there does seem to be a way with "refresh tokens", gonna have to investigate

**Comment by MrBrax** [comment]
i think it might work now... maybe?
